modules/aws-eks-iam-role-group/main.tf

```
@@ -35,6 +49,33 @@ resource "aws_iam_role" "iam_role" {
   assume_role_policy = data.aws_iam_policy_document.assume_role_policy.json
 }
 
+data "aws_iam_policy_document" "role_policy_for_cluster" {
+  statement {
+    sid     = "ClusterInfo"
+    effect  = "Allow"
+    actions = [
+      "eks:DescribeCluster",
+      "eks:ListClusters"
+    ]
+    resources = ["*"]
+  }
+  statement {
+    sid       = "AssumeRole"
+    effect    = "Allow"
+    actions   = ["sts:AssumeRole"]
+    resources = [aws_iam_role.iam_role.arn]
+  }
+}
+
+resource "aws_iam_role_policy" "role_access_policy" {
+  count = length(var.external_arn_role) == 0 ? 0 : 1
+
+  name = "AssumeRole_access"
+  role = aws_iam_role.iam_role.id
+
+  policy = data.aws_iam_policy_document.role_policy_for_cluster.json
+}
+
```

I think it's ok to add EKS permissions to the role regardless of `external_arn_role` presence. Also, we should grant this role the same permissions as we grant group members (role can not be a part of group, so we just have to attach it as in my suggestion below:

```suggestion
resource "aws_iam_role_policy" "role_eks_access" {
  name  = "EKS"
  role  = aws_iam_role.iam_role.id
  policy = data.aws_iam_policy_document.eks_access.json
}

resource "aws_iam_role_policy_attachment" "role_policies" {
  for_each = var.iam_group_policies # We may want to rename it to `iam_policies`
  
  role = aws_iam_role.iam_role.id
  policy = each.value
}

```